### ISSUE TITLE
Remove `getindex` and `setindex!`

### ISSUE THREAD

**Issue by jkozdon** [climate-machine/CLIMA#47]
Switch to using functions and not `getindex` and `setindex!` in the AtmosDycore

```julia
time(runner)
settime!(runner, time)
```

instead of

```julia
runner[:time]
runner[:time] = time
```

**Comment by jkozdon** [comment]
The Julian style is that really `getindex` and `setindex!` should be for collection type accesses, not necessarily to access members of a struct. I was trying to hide the implementation of things like `time` from the user, so did not want to assume that `runner.time` was valid. Another option would be to use `getproperty`

**Comment by jkozdon** [comment]
fixed by #54
